crates/orderbook-commons/src/lib.rs

```diff
@@ -351,15 +353,38 @@ pub fn get_filled_with_from_matches(matches: Vec<Matches>) -> Result<FilledWith>
     })
 }
 
-/// todo(holzeis): this should come from a configuration https://github.com/get10101/10101/issues/1029
-pub fn get_expiry_timestamp(from: OffsetDateTime) -> OffsetDateTime {
-    let tomorrow = from.date() + Duration::days(7);
-    tomorrow.midnight().assume_utc()
+/// Calculates the expiry timestamp at the next Sunday at 3 pm UTC from a given offset date time.
+/// If the argument falls in between Friday, 3 pm UTC and Sunday, 3pm UTC, the expiry will be
+/// calculated to next weeks Sunday at 3 pm
+pub fn get_expiry_timestamp(time: OffsetDateTime) -> OffsetDateTime {
+    let days = if is_in_rollover_weekend(time) || time.weekday() == Weekday::Sunday {
```

Comment:
because I have to handle the case if its Sunday 15:01. In that case its not the `rollover_weekend` anymore, but we still need to add 7 days to get to the next Sunday. 

`7 - time.weekday().number_from_monday() = 0`

Reply:
Okay, I get it now. A bit confusing nonetheless. Might be worth moving part of the comment below to the condition.

Reply:
🤔 what comment are you referring to?

Reply:
I don't think it makes it any better, so you can ignore my previous comment.